### PR TITLE
Updating SARIF SDK, removing the old schema, removing fixed version.

### PR DESCRIPTION
**Pull request by eddynaka**
- [x] You've read the [Contributor Guide](../CONTRIBUTING.md) and [Code of Conduct](../CODE-OF-CONDUCT.md).
- [x] You've included unit or integration tests for your change, where applicable.
- [ ] You've included inline docs for your change, where applicable.
- [ ] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.

**PR Title**
Updating SARIF SDK, removing the old schema, removing fixed version.

**PR Description**
- Updating to the latest SARIF.SDK version
- Removing hard-coded schema and version
- This will fix the load issue in VS Code

Addresses #bugnumber (in this specific format)

### CODE REVIEW

File: src/components/Microsoft.DotNet.UpgradeAssistant.Analysis/AnalyzeResultWriter.cs [hunk 1]
```diff
@@ -30,8 +30,6 @@ public async Task WriteAsync(IAsyncEnumerable<AnalyzeResultDefinition> results,
 
             var sarifLog = new SarifLog()
             {
-                Version = SarifVersion.Current,
-                SchemaUri = new Uri("http://json.schemastore.org/sarif-2.1.0"),
```
Comment (eddynaka) [inline src/components/Microsoft.DotNet.UpgradeAssistant.Analysis/AnalyzeResultWriter.cs:34]: When we create a SarifLog, by default it points to the Current version and the correct schema. With that, we don't need to change it.

File: tests/components/Microsoft.DotNet.UpgradeAssistant.Tests/AnalyzeResultWriterTests.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+using Microsoft.CodeAnalysis.Sarif;
+using Microsoft.DotNet.UpgradeAssistant.Analysis;
+using Xunit;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Tests
+{
+    public class AnalyzeResultWriterTests
+    {
+        [Fact]
+        public async Task ValidateSarifVersion()
```
Comment (eddynaka) [inline tests/components/Microsoft.DotNet.UpgradeAssistant.Tests/AnalyzeResultWriterTests.cs:19]: This is a simple E2E test that will create a SarifLog using the writer, load it back and we are just checking the version and schema to see if they are what we expect.

**Comment by eddynaka** [comment]
@sunandabalu as we talked offline, here you can find a few improvements in the SARIF. This also fix the VSCode load issue.

File: tests/components/Microsoft.DotNet.UpgradeAssistant.Tests/AnalyzeResultWriterTests.cs (hunk 2 above)
Comment (sunandabalu) [reply tests/components/Microsoft.DotNet.UpgradeAssistant.Tests/AnalyzeResultWriterTests.cs:19]: Awesome, to follow the code structure in Upgrade-Assistant the AnalyzeResultWriterTests should go under tests/components/Microsoft.DotNet.UpgradeAssistant.Analysis/

**Comment by eddynaka** [comment]
Below you can find the logs of a manual run:
```
[21:51:16 INF] Using MSBuild from C:\Program Files\dotnet\sdk\5.0.403\
[21:51:16 INF] Using Visual Studio install from C:\Program Files\Microsoft Visual Studio\2022\Preview [v17]
[21:51:29 INF] Reference to .NET Upgrade Assistant analyzer package (Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers, version 0.3.255803) needs to be added
[21:51:32 INF] Marking package Microsoft.Extensions.DependencyInjection.Abstractions for removal because it appears to be a transitive dependency
[21:51:32 INF] Marking package Microsoft.Bcl.AsyncInterfaces for removal because it appears to be a transitive dependency
[21:51:32 FTL] Package analysis failed (analyzer Target compatibility reference analyzer: Package references must have specific versions to get package archives (Parameter 'packageReference')
[21:51:32 INF] Reference to .NET Upgrade Assistant analyzer package (Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers, version 0.3.255803) needs to be added
[21:51:32 ERR] Package analysis failed
[21:51:32 INF] Running analyzers on Microsoft.DotNet.UpgradeAssistant.Analysis
[21:51:35 INF] Identified 0 diagnostics in project Microsoft.DotNet.UpgradeAssistant.Analysis
[21:51:35 INF] Running analyzers on Microsoft.DotNet.UpgradeAssistant.Abstractions
[21:51:36 INF] Identified 0 diagnostics in project Microsoft.DotNet.UpgradeAssistant.Abstractions
```

Attached the result :)
[AnalysisReport.zip](https://github.com/dotnet/upgrade-assistant/files/7516844/AnalysisReport.zip)

File: tests/components/Microsoft.DotNet.UpgradeAssistant.Tests/Microsoft.DotNet.UpgradeAssistant.Tests.csproj [hunk 3]
```diff
@@ -3,6 +3,7 @@
     <TargetFramework>net5.0</TargetFramework>
   </PropertyGroup>
   <ItemGroup>
+    <ProjectReference Include="..\..\..\src\components\Microsoft.DotNet.UpgradeAssistant.Analysis\Microsoft.DotNet.UpgradeAssistant.Analysis.csproj" />
```
Comment (sunandabalu) [inline tests/components/Microsoft.DotNet.UpgradeAssistant.Tests/Microsoft.DotNet.UpgradeAssistant.Tests.csproj:6]: I'm guess this is not needed here anymore?

Comment (eddynaka) [reply tests/components/Microsoft.DotNet.UpgradeAssistant.Tests/Microsoft.DotNet.UpgradeAssistant.Tests.csproj:6]: removed :)

File: tests/components/Microsoft.DotNet.UpgradeAssistant.Tests/AnalyzeResultWriterTests.cs (hunk 2 above)
Comment (eddynaka) [reply tests/components/Microsoft.DotNet.UpgradeAssistant.Tests/AnalyzeResultWriterTests.cs]: created the new project and moved the file

**Comment by sunandabalu** [comment]
> Below you can find the logs of a manual run:
> 
> ```
> [21:51:16 INF] Using MSBuild from C:\Program Files\dotnet\sdk\5.0.403\
> [21:51:16 INF] Using Visual Studio install from C:\Program Files\Microsoft Visual Studio\2022\Preview [v17]
> [21:51:29 INF] Reference to .NET Upgrade Assistant analyzer package (Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers, version 0.3.255803) needs to be added
> [21:51:32 INF] Marking package Microsoft.Extensions.DependencyInjection.Abstractions for removal because it appears to be a transitive dependency
> [21:51:32 INF] Marking package Microsoft.Bcl.AsyncInterfaces for removal because it appears to be a transitive dependency
> [21:51:32 FTL] Package analysis failed (analyzer Target compatibility reference analyzer: Package references must have specific versions to get package archives (Parameter 'packageReference')
> [21:51:32 INF] Reference to .NET Upgrade Assistant analyzer package (Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers, version 0.3.255803) needs to be added
> [21:51:32 ERR] Package analysis failed
> [21:51:32 INF] Running analyzers on Microsoft.DotNet.UpgradeAssistant.Analysis
> [21:51:35 INF] Identified 0 diagnostics in project Microsoft.DotNet.UpgradeAssistant.Analysis
> [21:51:35 INF] Running analyzers on Microsoft.DotNet.UpgradeAssistant.Abstractions
> [21:51:36 INF] Identified 0 diagnostics in project Microsoft.DotNet.UpgradeAssistant.Abstractions
> ```
> 
> Attached the result :) [AnalysisReport.zip](https://github.com/dotnet/upgrade-assistant/files/7516844/AnalysisReport.zip)

Perfect, thanks for getting rid of the pesky warning of unsupported version type :)

**Review by sunandabalu — APPROVED** [review]
LGTM

**Pull request merged**